internal/engine/image_build_and_deployer.go

```diff
@@ -156,14 +156,20 @@ func (ibd *ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.R
 		anyLiveUpdate = anyLiveUpdate || !iTarget.LiveUpdateInfo().Empty()
 		return store.NewImageBuildResult(iTarget.ID(), refs.LocalRef, refs.ClusterRef), nil
 	})
+
+	newResults := q.NewResults()
 	if err != nil {
-		return store.BuildResultSet{}, buildcontrol.WrapDontFallBackError(err)
+		return newResults, buildcontrol.WrapDontFallBackError(err)
```

Comment:
hmmm so this results set ought to only contain results for targets we actually built this time, IF we built them successfully, right? (just checking my understanding)

Reply:
yep